### ISSUE TITLE
Segmentation fault of vl_nnconv

### ISSUE THREAD

**Issue by denru01** [vlfeat/matconvnet#698]
I am running matconvnet on a server with 9 GPUs (all are identical) and I am using CUDA 8.0 (all the GPUs are pascal). I encountered a problem that I can successfully use the 1~8th GPUs but not 9th GPU.

The command "gpuDevice(9)" or other matlab native GPU functions worked correctly. However, when I run a matconvnet program, it always gave the error "The GPU failed to allocate memory. To continue, reset the GPU by running 'gpuDevice(9)'. If this problem persists, partition your computations into smaller pieces.". Running "gpuDevice(9)" did not fix the problem. 

The strange thing is that the gpuArray I would like to create is very small. I checked with "nvidia-smi" and there was lots of free GPU memory. Because all the 9 GPUs are the same, I tried to run the same program on another 8 GPUs, and it worked perfectly. 

After I used gpuDevice(9) to run the program, I cannot use any other GPUs. When I used something like "gpuDevice(1)", it gave the error "An unexpected error occurred during CUDA execution. The CUDA error was: all CUDA-capable devices are busy or unavailable". However, they were idle, not busy. The only way to solve this problem is restarting matlab and avoiding using gpuDevice(9) or matconvnet.

I tried to run programs on the 9th GPU by using python, and it worked correctly without any problem.
Sometimes when I closed matlab after I encountered this problem, matlab gave a system error (attached below). 

I tried this on both matconvnet-beta20 and beta22, and all leaded to the same result. Driver version is 367.35.

------------------------------------------------------------------------
       Segmentation violation detected at Mon Sep 19 15:40:20 2016
------------------------------------------------------------------------

Configuration:
  Crash Decoding      : Disabled
  Crash Mode          : continue (default)
  Current Graphics Driver: Unknown software 
  Current Visual      : 0x22 (class 4, depth 24)
  Default Encoding    : UTF-8
  GNU C Library       : 2.23 stable
  Host Name           : xxx
  MATLAB Architecture : glnxa64
  MATLAB Root         : /usr/local/MATLAB/R2016a
  MATLAB Version      : 9.0.0.341360 (R2016a)
  OpenGL              : software
  Operating System    : Linux 4.4.0-36-generic #55-Ubuntu SMP Thu Aug 11 18:01:55 UTC 2016 x86_64
  Processor ID        : x86 Family 6 Model 79 Stepping 1, GenuineIntel
  Virtual Machine     : Java 1.7.0_60-b19 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  Window System       : AT&T Laboratories Cambridge (3332), display :7

Fault Count: 1

Abnormal termination:
Segmentation violation

Register State (from fault):
  RAX = 0000000000000000  RBX = 00007f185e5939f0
  RCX = 00007f1854886610  RDX = 00007f18cd028ac0
  RSP = 00007f1a06c5fd98  RBP = 00007f185e5939f0
  RSI = 0000000000000000  RDI = cc006400171ce066

   R8 = 0000000000000000   R9 = 0000000000000000
  R10 = 000000000000002c  R11 = 0000000000000025
  R12 = 00007f1855da2220  R13 = 0000000000000000
  R14 = 00007f185e5939f0  R15 = 00007f1a007f19a0

  RIP = 00007f18cc7fd2da  EFL = 0000000000010286

   CS = 0033   FS = 0000   GS = 0000

Stack Trace (from fault):
[  0] 0x00007f18cc7fd2da             /usr/lib/x86_64-linux-gnu/libcuda.so.1+01655514
[  1] 0x00007f18cc724996             /usr/lib/x86_64-linux-gnu/libcuda.so.1+00768406
[  2] 0x00007f18cc854ea2             /usr/lib/x86_64-linux-gnu/libcuda.so.1+02014882 cuEventDestroy_v2+00000082
[  3] 0x00007f17f1005254            /usr/local/cuda-8.0/lib64/libcudnn.so.5+05182036
[  4] 0x00007f17f103f5e4            /usr/local/cuda-8.0/lib64/libcudnn.so.5+05420516
[  5] 0x00007f17f0b4b965            /usr/local/cuda-8.0/lib64/libcudnn.so.5+00227685 cudnnDestroy+00000149
[  6] 0x00007f17e215b45f /home/tjy/Libraries/matconvnet-1.0-beta20-CompCap3.7/matlab/mex/vl_nnconv.mexa64+00234591
[  7] 0x00007f17e215b30e /home/tjy/Libraries/matconvnet-1.0-beta20-CompCap3.7/matlab/mex/vl_nnconv.mexa64+00234254
[  8] 0x00007f17e215b269 /home/tjy/Libraries/matconvnet-1.0-beta20-CompCap3.7/matlab/mex/vl_nnconv.mexa64+00234089
[  9] 0x00007f17e212976d /home/tjy/Libraries/matconvnet-1.0-beta20-CompCap3.7/matlab/mex/vl_nnconv.mexa64+00030573
[ 10] 0x00007f17e2127647 /home/tjy/Libraries/matconvnet-1.0-beta20-CompCap3.7/matlab/mex/vl_nnconv.mexa64+00022087
[ 11] 0x00007f1a1445d966     /usr/local/MATLAB/R2016a/bin/glnxa64/libmex.so+00149862
[ 12] 0x00007f1a1445da46     /usr/local/MATLAB/R2016a/bin/glnxa64/libmex.so+00150086
[ 13] 0x00007f1a1445a5ec     /usr/local/MATLAB/R2016a/bin/glnxa64/libmex.so+00136684
[ 14] 0x00007f1a1445aab8     /usr/local/MATLAB/R2016a/bin/glnxa64/libmex.so+00137912
[ 15] 0x00007f1a15205095 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00749717 _ZN8Mlm_file9unload_mfEv+00000021
[ 16] 0x00007f1a151b6b3b /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00428859 _ZN13Mlm_MATLAB_fn6unloadEv+00000107
[ 17] 0x00007f1a151b7409 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00431113 _ZN13Mlm_MATLAB_fn5clearEv+00000041
[ 18] 0x00007f1a151f5c40 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00687168
[ 19] 0x00007f1a151f452e /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00681262 mdClearFunctionsByName+00000190
[ 20] 0x00007f1a1213f56e /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08508782
[ 21] 0x00007f1a1213f651 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08509009
[ 22] 0x00007f1a1213f6e9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08509161
[ 23] 0x00007f1a1213f97c /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08509820
[ 24] 0x00007f1a1213faa5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08510117
[ 25] 0x00007f1a151c33f5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00480245
[ 26] 0x00007f1a151ab736 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00382774 _ZN13Mfh_MATLAB_fn11dispatch_fhEiPP11mxArray_tagiS2_+00000390
[ 27] 0x00007f1a121f01c9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+09232841
[ 28] 0x00007f1a1231cdbf /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10464703
[ 29] 0x00007f1a12312a5a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10422874
[ 30] 0x00007f1a122db911 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10197265
[ 31] 0x00007f1a11da9b2a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04750122
[ 32] 0x00007f1a11daaa4c /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04753996
[ 33] 0x00007f1a11da8ebc /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04746940
[ 34] 0x00007f1a11da69ea /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04737514
[ 35] 0x00007f1a11da6db1 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04738481
[ 36] 0x00007f1a11da8a63 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04745827
[ 37] 0x00007f1a11da8be9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+04746217
[ 38] 0x00007f1a11e5916f /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+05468527
[ 39] 0x00007f1a11e5c22a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+05481002
[ 40] 0x00007f1a120fe543 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08242499
[ 41] 0x00007f1a120c707e /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08015998
[ 42] 0x00007f1a120cb058 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08032344
[ 43] 0x00007f1a120cb107 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08032519
[ 44] 0x00007f1a1214abf4 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08555508
[ 45] 0x00007f1a1214b251 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08557137
[ 46] 0x00007f1a151c33f5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00480245
[ 47] 0x00007f1a151ab736 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00382774 _ZN13Mfh_MATLAB_fn11dispatch_fhEiPP11mxArray_tagiS2_+00000390
[ 48] 0x00007f1a1235d42e /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10728494
[ 49] 0x00007f1a1235d4c9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10728649
[ 50] 0x00007f1a120c5e1c /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08011292
[ 51] 0x00007f1a1214045a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08512602
[ 52] 0x00007f1a15505827   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00792615 _ZN11mcrInstance27feval_on_interpreter_threadEPKciPP11mxArray_tagiS4_+00000103
[ 53] 0x00007f1a15508e49   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00806473 _ZN11mcrInstance18CallMATLABFunctionEPKciPP11mxArray_tagiS4_PNSt15__exception_ptr13exception_ptrE+00000121
[ 54] 0x00007f1a155090a9   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00807081 _ZN11mcrInstance10EvalStringERKSbIDsSt11char_traitsIDsESaIDsEEiPP11mxArray_tag+00000137
[ 55] 0x00007f1a155091c9   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00807369 _ZN11mcrInstance14clearAllOnExitEv+00000153
[ 56] 0x00007f1a1552198a   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00907658
[ 57] 0x00007f1a15521dd1   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00908753
[ 58] 0x00007f1a15521ebb   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00908987
[ 59] 0x00007f1a15522104   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00909572
[ 60] 0x00007f1a151c33f5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00480245
[ 61] 0x00007f1a151ab736 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00382774 _ZN13Mfh_MATLAB_fn11dispatch_fhEiPP11mxArray_tagiS2_+00000390
[ 62] 0x00007f1a0543969a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcos_impl.so+02705050
[ 63] 0x00007f1a151f50ff /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00684287
[ 64] 0x00007f1a151c33f5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00480245
[ 65] 0x00007f1a151ab736 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_dispatcher.so+00382774 _ZN13Mfh_MATLAB_fn11dispatch_fhEiPP11mxArray_tagiS2_+00000390
[ 66] 0x00007f1a1235d42e /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10728494
[ 67] 0x00007f1a1235d4c9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+10728649
[ 68] 0x00007f1a120c5e1c /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08011292
[ 69] 0x00007f1a120c63ba /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08012730
[ 70] 0x00007f1a12140544 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08512836
[ 71] 0x00007f1a1214021d /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08512029
[ 72] 0x00007f1a1214101f /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_lxe.so+08515615
[ 73] 0x00007f1a14bc51f0 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwm_interpreter.so+03166704 _Z44inCallFcnWithTrapInDesiredWSAndPublishEventsiPP11mxArray_tagiS1_PKcbP15inWorkSpace_tag+00000096
[ 74] 0x00007f1a1583c1b4   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00819636 _ZN3iqm15BaseFEvalPlugin7executeEP15inWorkSpace_tagRN5boost10shared_ptrIN14cmddistributor17IIPCompletedEventEEE+00000356
[ 75] 0x00007f19b7599fdf /usr/local/MATLAB/R2016a/bin/glnxa64/libnativejmi.so+00966623 _ZN9nativejmi14JmiFEvalPlugin7executeEP15inWorkSpace_tagRN5boost10shared_ptrIN14cmddistributor17IIPCompletedEventEEE+00000335
[ 76] 0x00007f19b75c1c0d /usr/local/MATLAB/R2016a/bin/glnxa64/libnativejmi.so+01129485 _ZN3mcr3mvm27McrSwappingIqmPluginAdapterIN9nativejmi14JmiFEvalPluginEE7executeEP15inWorkSpace_tagRN5boost10shared_ptrIN14cmddistributor17IIPCompletedEventEEE+00000429
[ 77] 0x00007f1a15830f98   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00774040
[ 78] 0x00007f1a1581ac87   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00683143
[ 79] 0x00007f1a146927a9 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwbridge.so+00173993
[ 80] 0x00007f1a14692c44 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwbridge.so+00175172
[ 81] 0x00007f1a146991a5 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwbridge.so+00201125
[ 82] 0x00007f1a146992ac /usr/local/MATLAB/R2016a/bin/glnxa64/libmwbridge.so+00201388
[ 83] 0x00007f1a1469997b /usr/local/MATLAB/R2016a/bin/glnxa64/libmwbridge.so+00203131 _Z8mnParserv+00000715
[ 84] 0x00007f1a154edc13   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00695315
[ 85] 0x00007f1a2517581c   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmvm.so+02410524 _ZNK5boost9function0IvEclEv+00000028
[ 86] 0x00007f1a154ef925   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00702757
[ 87] 0x00007f1a154f0839   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00706617 _ZN5boost6detail17task_shared_stateINS_3_bi6bind_tIvPFvRKNS_8functionIFvvEEEENS2_5list1INS2_5valueIS6_EEEEEEvE6do_runEv+00000025
[ 88] 0x00007f1a1f8fe8a0    /usr/local/MATLAB/R2016a/bin/glnxa64/libmwms.so+03168416 _ZN5boost6detail22task_base_shared_stateIvE3runEv+00000064
[ 89] 0x00007f1a154f1717   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00710423
[ 90] 0x00007f1a154ed75a   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00694106
[ 91] 0x00007f1a15860c06   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00969734
[ 92] 0x00007f1a1584eb4c   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00895820 _ZN5boost6detail8function21function_obj_invoker0ISt8functionIFNS_3anyEvEES4_E6invokeERNS1_15function_bufferE+00000028
[ 93] 0x00007f1a1584f21f   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00897567 _ZNK5boost9function0INS_3anyEEclEv+00000031
[ 94] 0x00007f1a1584e993   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00895379 _ZN3iqm18PackagedTaskPlugin7executeEP15inWorkSpace_tagRN5boost10shared_ptrIN14cmddistributor17IIPCompletedEventEEE+00000163
[ 95] 0x00007f1a1550271d   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00780061
[ 96] 0x00007f1a15830f98   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00774040
[ 97] 0x00007f1a1581b5af   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00685487
[ 98] 0x00007f1a158187e3   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwiqm.so+00673763
[ 99] 0x00007f1a24b32a3a /usr/local/MATLAB/R2016a/bin/glnxa64/libmwservices.so+03443258
[100] 0x00007f1a24b318a7 /usr/local/MATLAB/R2016a/bin/glnxa64/libmwservices.so+03438759
[101] 0x00007f1a24b3210c /usr/local/MATLAB/R2016a/bin/glnxa64/libmwservices.so+03440908 _Z25svWS_ProcessPendingEventsiib+00000092
[102] 0x00007f1a154eded2   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00696018
[103] 0x00007f1a154ee211   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00696849
[104] 0x00007f1a154dacfd   /usr/local/MATLAB/R2016a/bin/glnxa64/libmwmcr.so+00617725
[105] 0x00007f1a2330e6fa              /lib/x86_64-linux-gnu/libpthread.so.0+00030458
[106] 0x00007f1a23044b5d                    /lib/x86_64-linux-gnu/libc.so.6+01076061 clone+00000109
[107] 0x0000000000000000                                   <unknown-module>+00000000

**Comment by lenck** [comment]
Hmm, we do not have any machine with more than 4 GPUs so I have no idea what this issue can be. I would think it's a problem of MATLAB, when it crashes when you try to close it.
Can you test if some of the MATLAB [GPU Mex examples](http://uk.mathworks.com/help/distcomp/run-mex-functions-containing-cuda-code.html) work?

**Comment by denru01** [comment]
Yes, it works.

**Comment by lyltencent** [comment]
Hi @lenck , I have the similar problem when I am running the code. 
My matlab crashes during the training of CNN. I have to restart matlab and start pick up the latest net-epoch-XXX.mat and continue training from there. 
I s this problem being solved ? @denru01 

Thanks,

**Comment by putama** [comment]
has this issue been solved? I encountered the exactly same error even with only a single GPU.

**Comment by denru01** [comment]
@lyltencent This problem still exists. I am using 1.0-beta24.

**Comment by oakal** [comment]
Had a similar issue while running CNN, all of a sudden in the middle matlab was giving me this error let say at 30th epoch then reset the matlab and start where left of  it gives the same error let say around 50th epoch. Found a solution for this like a month ago i don't remember what was the reasoning but if you open matlab with softwareopengl you shouldn't have this problem again. basically use the following command to start matlab

$ matlab -softwareopengl

BTW: I'm using MatConvNet 1.0-beta24 with matlab R2017a, cuda 8.0 and cudnn 5.1 on GTX 1080 ubuntu 16.04 machine

**Comment by GuoShi28** [comment]
I meet familiar problem and I do not know how to fix it.
------------------------------------------------------------------------
       Segmentation violation detected at Wed Jul 18 22:09:32 2018
------------------------------------------------------------------------

Configuration:
  Crash Decoding      : Disabled - No sandbox or build area path
  Crash Mode          : continue (default)
  Current Graphics Driver: NVIDIA Corporation GeForce GTX 1080 Ti/PCIe/SSE2 Version 4.6.0 NVIDIA 390.77
  Current Visual      : 0x21 (class 4, depth 24)
  Default Encoding    : UTF-8
  Deployed            : false
  GNU C Library       : 2.23 stable
  Host Name           : guoshi-polyu
  MATLAB Architecture : glnxa64
  MATLAB Entitlement ID: 6257193
  MATLAB Root         : /usr/local/MATLAB/R2017a
  MATLAB Version      : 9.2.0.538062 (R2017a)
  OpenGL              : hardware
  Operating System    : Linux 4.13.0-45-generic #50~16.04.1-Ubuntu SMP Wed May 30 11:18:27 UTC 2018 x86_64
  Processor ID        : x86 Family 6 Model 63 Stepping 2, GenuineIntel
  Virtual Machine     : Java 1.7.0_60-b19 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  Window System       : The X.Org Foundation (11905000), display :0

Fault Count: 1


Abnormal termination:
Segmentation violation

Register State (from fault):
  RAX = 0000000000000001  RBX = 00007f39fea252e0
  RCX = 0000000000000003  RDX = 00000000ffffffff
  RSP = 00007f3a2c15a210  RBP = 00007f3a2c15a210
  RSI = 00007f3a2c15a290  RDI = 00007f39fea252e0

   R8 = 00007f39fc5789b0   R9 = 00007f387176dca0
  R10 = 0000000000000000  R11 = 0000000000000000
  R12 = 00007f3a2c15a310  R13 = 00007f387176dca0
  R14 = 00007f3a2c15a290  R15 = 0000000000000000

  RIP = 00007f389084f198  EFL = 0000000000010202

   CS = 0033   FS = 0000   GS = 0000

Stack Trace (from fault):
[  0] 0x00007f389084f198                      bin/glnxa64/libmwosgserver.so+02630040 _ZN2hg22SoftwareClippingValues22setOriginalClipPlaneEqERN3osg7ref_ptrINS1_13TemplateArrayINS1_5Vec4fELNS1_5Array4TypeE29ELi4ELi5126EEEEE+00000040
[  1] 0x00007f3890844d19                      bin/glnxa64/libmwosgserver.so+02587929 _ZN2hg14openscenegraph27setSoftwareClippingUserDataEPN3osg5GeodeEPN7osgUtil11CullVisitorENS_17SoftwareClipStyleE+00000969
[  2] 0x00007f389077e791                      bin/glnxa64/libmwosgserver.so+01775505
[  3] 0x00007f3890839d4c                      bin/glnxa64/libmwosgserver.so+02542924 _ZN2hg14openscenegraph23OsgGeometryCullCallbackclEPN3osg4NodeEPNS2_11NodeVisitorE+00000636
[  4] 0x00007f38713b00de /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01212638 _ZN3osg12NodeCallback3runEPNS_6ObjectES2_+00000094
[  5] 0x00007f386fb8dd13 /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00908563 _ZN7osgUtil11CullVisitor5applyERN3osg5GeodeE+00000227
[  6] 0x00007f38713eaf98 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01453976 _ZN3osg5Geode6acceptERNS_11NodeVisitorE+00000120
[  7] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[  8] 0x00007f386fb8d78b /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00907147 _ZN7osgUtil11CullVisitor5applyERN3osg5GroupE+00000219
[  9] 0x00007f387140e418 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01598488 _ZN3osg5Group6acceptERNS_11NodeVisitorE+00000120
[ 10] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 11] 0x00007f386fb8d78b /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00907147 _ZN7osgUtil11CullVisitor5applyERN3osg5GroupE+00000219
[ 12] 0x00007f387140e418 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01598488 _ZN3osg5Group6acceptERNS_11NodeVisitorE+00000120
[ 13] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 14] 0x00007f386fb8d78b /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00907147 _ZN7osgUtil11CullVisitor5applyERN3osg5GroupE+00000219
[ 15] 0x00007f387140e418 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01598488 _ZN3osg5Group6acceptERNS_11NodeVisitorE+00000120
[ 16] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 17] 0x00007f38713afe9b /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01212059 _ZN3osg8Callback8traverseEPNS_6ObjectES2_+00000123
[ 18] 0x00007f38907b9a4d                      bin/glnxa64/libmwosgserver.so+02017869 _ZN2hg24OsgDataSpaceCullCallbackclEPN3osg4NodeEPNS1_11NodeVisitorE+00000061
[ 19] 0x00007f38713b00de /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01212638 _ZN3osg12NodeCallback3runEPNS_6ObjectES2_+00000094
[ 20] 0x00007f386fb8f056 /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00913494 _ZN7osgUtil11CullVisitor5applyERN3osg9TransformE+00000198
[ 21] 0x00007f38714451a8 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01823144 _ZN3osg15MatrixTransform6acceptERNS_11NodeVisitorE+00000120
[ 22] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 23] 0x00007f386fb8d78b /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00907147 _ZN7osgUtil11CullVisitor5applyERN3osg5GroupE+00000219
[ 24] 0x00007f387140e418 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01598488 _ZN3osg5Group6acceptERNS_11NodeVisitorE+00000120
[ 25] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 26] 0x00007f386fb8e858 /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00911448 _ZN7osgUtil11CullVisitor5applyERN3osg6CameraE+00002632
[ 27] 0x00007f389080aa9c                      bin/glnxa64/libmwosgserver.so+02349724 _ZN3osg6Camera6acceptERNS_11NodeVisitorE+00000092
[ 28] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 29] 0x00007f386fb8d78b /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+00907147 _ZN7osgUtil11CullVisitor5applyERN3osg5GroupE+00000219
[ 30] 0x00007f387140e418 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01598488 _ZN3osg5Group6acceptERNS_11NodeVisitorE+00000120
[ 31] 0x00007f387140cc63 /usr/local/MATLAB/R2017a/bin/glnxa64/libosg.so.130+01592419 _ZN3osg5Group8traverseERNS_11NodeVisitorE+00000051
[ 32] 0x00007f386fc36bd7 /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+01600471 _ZN7osgUtil9SceneView9cullStageERKN3osg7MatrixdES4_PNS_11CullVisitorEPNS_10StateGraphEPNS_11RenderStageEPNS1_8ViewportE+00001799
[ 33] 0x00007f386fc34c40 /usr/local/MATLAB/R2017a/bin/glnxa64/libosgUtil.so.130+01592384 _ZN7osgUtil9SceneView4cullEv+00000960
[ 34] 0x00007f3890823a72                      bin/glnxa64/libmwosgserver.so+02452082 _ZN17SceneRendererImpl11renderSceneERN2hg14openscenegraph11RenderTimesERNS1_12RenderCountsEPN7osgUtil9SceneViewEj+00000066
[ 35] 0x00007f38908214da                      bin/glnxa64/libmwosgserver.so+02442458 _ZN2hg14openscenegraph14OsgSceneServer11cullAndDrawEibdd+00003722
[ 36] 0x00007f39ee6226d4                              bin/glnxa64/libuij.so+00493268 _Z23UIJ_call_OpenGLPaintFcnPN11gui_objects12scene_server16SceneServerProxyEibdd+00000020
[ 37] 0x00007f39b717ce1e                         bin/glnxa64/libnativehg.so+00044574 Java_com_mathworks_hg_peer_JavaSceneServerPeer_doDisplay+00000030
[ 38] 0x00007f3a71012d98                                   <unknown-module>+00000000
[ 39] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 40] 0x00007f3a71006706                                   <unknown-module>+00000000
[ 41] 0x00007f3a71006706                                   <unknown-module>+00000000
[ 42] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 43] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 44] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 45] 0x00007f3a71006706                                   <unknown-module>+00000000
[ 46] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 47] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 48] 0x00007f3a71006706                                   <unknown-module>+00000000
[ 49] 0x00007f3a71006058                                   <unknown-module>+00000000
[ 50] 0x00007f3a711e3e64                                   <unknown-module>+00000000
